src/components/Organization/EditOrganizationModal/EditOrgModal.tsx

```diff
@@ -0,0 +1,169 @@
+'use client';
+// CSS imports
+import { createSupbaseClient } from '@/lib/supabase/client';
+import { IOrgEdit } from '@/types/componentTypes';
+import { IOrganization } from '@/types/database.interface';
+import Image from 'next/image';
+import { useRouter } from 'next/navigation';
+import { useState } from 'react';
+import './EditOrgModal.css';
+
+export default function EditOrgModal({
+	clickHandler,
+	organization,
+}: {
+	clickHandler: () => void;
+	organization: IOrgEdit;
+}) {
+	const [formData, setFormData] = useState({
+		orgName: organization.name,
+		orgImageUrl: organization.image,
+		orgImage: useState<any>([]),
+	});
+
+	const DEFAULT_IMAGE =
+		'https://apqmqmysgnkmkyesdrnn.supabase.co/storage/v1/object/public/profile-avatars/wyncoservices.png';
+	const PREVIOUS_IMAGE = organization.image;
+	console.log(PREVIOUS_IMAGE);
+	const router = useRouter();
+
+	const [image, setImage] = useState<any>([]);
+	const [imageURL, setImageURL] = useState(PREVIOUS_IMAGE);
+	const [orgError, setOrgError] = useState('');
+	var isNewImage = false;
+
+	const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+		setFormData({
+			...formData,
+			[e.target.id]: e.target.value,
+		});
+	};
+
+	const handleImageChange = (e: any) => {
+		if (!e.target.files) {
+			return;
+		}
+		isNewImage = true;
+
+		setImage(e.target.files[0]);
+		setImageURL(URL.createObjectURL(e.target.files[0]));
+	};
+
+	const handleSubmit = async (e: any) => {
+		// upload image
+		await editOrganization(e);
+	};
+
+	const editOrganization = async (e: any) => {
```

Comment:
Once implemented to main branch, lets move such functions to a separate dedicated file under the `lib/actions` folder likely, and subfolder if you want giving it whatever name you find fitting.